spylib/request.py

```diff
@@ -129,9 +134,24 @@ def make_service_request(
         else:
             raise MethodException
 
-        UNKNOWN_SERVER_ERROR = 500
+        # Check for a credential failure - if so, cycle our tokens and try again w/ no retry
+        # Note: We pass a negative retry_count here to prevent an infinite chain
+        if resp.status_code in [401] and retry_count >= 0:
+            self.fetch_new_tokens()
+            second_resp = self.make_service_request(
```

Comment:
We could just return here to simplify the number of `resp` objects laying around

Reply:
Good call! I'll do that.